art/attacks/evasion/fast_gradient.py

```diff
@@ -102,6 +102,21 @@ def __init__(
         self._project = True
         FastGradientMethod._check_params(self)
 
+    def _check_compatibility_input_and_eps(self, x: np.ndarray):
+        """
+        Check the compatibility of the input with `eps` and `eps_step` which are of the same shape.
+
+        :param x: An array with the original inputs.
+        """
+        if isinstance(self.eps, np.ndarray):
+            if isinstance(self.estimator, ClassifierMixin):
+                # Ensure the eps array is broadcastable
+                if len(self.eps.shape) > len(x.shape) or self.eps.shape != x.shape[-len(self.eps.shape) :]:
```

Comment:
I'm not sure about the right-hand side of the `or` because it does not allow `eps` of shape `(1, 4)` for `x` of shape `(n, 4)` although Numpy broadcasting rules allow it.

Reply:
Fixed.